include/OpenSpaceToolkit/Astrodynamics/Flight/Profile.hpp

```diff
@@ -173,10 +217,52 @@ class Profile
         const trajectory::Orbit& anOrbit, const trajectory::Orbit::FrameType& anOrbitalFrameType
     );
 
+    /// @brief Construct a flight profile with custom target pointing
+    ///
+    /// @param anOrbit An orbit
+    /// @param anOrientationGenerator
+    /// @return Flight profile
+    static Profile GenerateTrackingProfile(
+        const trajectory::Orbit& anOrbit, const std::function<Quaternion(const State&)>& anOrientationGenerator
+    );
+
+    /// @brief Generate a custom orientation
+    ///
+    /// @param anAlignmentAxis An alignment axis
+    /// @param aClockingAxis A clocking axis
+    /// @param anAngularOffset An angular offset
+    /// @param aTrajectory A trajectory
+
+    static std::function<Quaternion(const State&)> GenerateCustomOrientation(
+        const Target& anAlignmentTarget,
+        const Target& aClockingTarget,
+        const Angle& anAngularOffset = Angle::Zero(),
+        const Trajectory& aTrajectory = Trajectory::Undefined()
```

Comment:
What if we supplied two Targets with `type = Trajectory`? I think this should be associated with the Target struct itself to avoid the potential confusion.